common/views/components/CatalogueSearchContext/CatalogueSearchContext.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+import {
+  type Node,
+  createContext,
+  useState,
+  useContext,
+  useEffect,
+} from 'react';
+import Router, { withRouter } from 'next/router';
+import { worksUrl } from '@weco/common/services/catalogue/urls';
+
+export type CatalogueQuery = {|
+  query: string,
+  page: number,
+  workType: ?(string[]),
+  itemsLocationsLocationType: ?(string[]),
+  queryType: ?string,
+|};
+
+type ContextProps = {|
+  ...CatalogueQuery,
+  setQuery: (value: string) => void,
+  setPage: (value: number) => void,
+  setWorkType: (value: ?(string[])) => void,
+  setItemsLocationsLocationType: (value: ?(string[])) => void,
+  setQueryType: (value: ?string) => void,
+|};
+
+const defaultState: CatalogueQuery = {
+  query: '',
+  page: 1,
+  workType: null,
+  itemsLocationsLocationType: null,
+  queryType: null,
+};
+
+const CatalogueSearchContext = createContext<ContextProps>({
+  ...defaultState,
+  setQuery: () => {},
+  setPage: () => {},
+  setWorkType: () => {},
+  setItemsLocationsLocationType: () => {},
+  setQueryType: () => {},
+});
+
+type CatalogueSearchProviderProps = {
+  children: Node,
+};
+
+const CatalogueSearchProvider = ({
+  children,
+  router,
+}: {
+  ...CatalogueSearchProviderProps,
+  router: any,
+}) => {
+  const initialState = {
+    query: router.query.query ? router.query.query : '',
+    page: router.query.page ? parseInt(router.query.page, 10) : 1,
+    workType: router.query.workType ? router.query.workType.split(',') : null,
+    itemsLocationsLocationType: router.query['items.locations.locationType']
+      ? router.query['items.locations.locationType'].split(',')
+      : null,
+  };
+  const state = {
+    ...defaultState,
+    ...initialState,
+  };
+
+  const [query, setQuery] = useState(state.query);
+  const [page, setPage] = useState(state.page);
+  const [workType, setWorkType] = useState(state.workType);
+  const [itemsLocationsLocationType, setItemsLocationsLocationType] = useState(
+    state.itemsLocationsLocationType
+  );
+  const [queryType, setQueryType] = useState(state.queryType);
+  const value = {
+    query,
+    page,
+    workType,
+    itemsLocationsLocationType,
+    queryType,
+    setQuery,
+    setPage,
+    setWorkType,
+    setItemsLocationsLocationType,
+    setQueryType,
+  };
+
+  useEffect(() => {
+    function routeChangeComplete(url: string) {
```

Comment:
Does this function have to live inside `useEffect`?

Reply:
This should only run on the client, so yip